server/src/main/java/io/crate/action/sql/Cursor.java

```diff
@@ -205,15 +213,20 @@ private void triggerConsumer(RowConsumer consumer, BatchIterator<Row> fullResult
         } else {
             int start = cursorPosition + count;
             assert start < cursorPosition : "count must be negative";
-            List<Object[]> items = Lists2.reverse(rows.subList(start - 1, cursorPosition - 1));
+            List<Object[]> items = Lists2.reverse(rows.subList(Math.max(start - 1, 0), Math.max(cursorPosition - 1, 0)));
             BatchIterator<Row> bi = biFromItems(items);
-            cursorPosition = start;
+            cursorPosition = Math.max(start, 0);
             consumer.accept(bi, null);
         }
     }
 
     private BatchIterator<Row> biFromItems(List<Object[]> items) {
-        BatchIterator<Object[]> objectRows = InMemoryBatchIterator.of(items, null, false);
+        BatchIterator<Object[]> objectRows;
+        if (items.isEmpty()) {
+            objectRows = InMemoryBatchIterator.empty(new Object[]{});
```

Comment:
```suggestion
            objectRows = InMemoryBatchIterator.empty(null);
```

In the `else` branch `null` is also used as sentinel. (and there is no need to create a new object array)